get_git.go

```diff
@@ -15,6 +15,10 @@ import (
 // a git repository.
 type GitGetter struct{}
 
+func (g *GitGetter) ClientMode(_ *url.URL) (ClientMode, error) {
+	return ClientModeDir, nil
```

Comment:
This seemed the best we could do for the 95% case in the VCS getters. Although we do have `GetFile()` for the git getter, it is not usable because of the way the source and dest are parsed out of the URL. It seems like it will take some thought to get it right in a generic way for `git` and `hg`.